modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/display/context/DLItemSelectorViewDisplayContext.java

```diff
@@ -364,8 +368,39 @@ private Long _getFileEntryTypeId() {
 		InfoItemItemSelectorCriterion infoItemItemSelectorCriterion =
 			(InfoItemItemSelectorCriterion)_itemSelectorCriterion;
 
-		return GetterUtil.getLong(
-			infoItemItemSelectorCriterion.getItemSubtype());
+		long fileEntryTypeId = GetterUtil.getLong(
+			infoItemItemSelectorCriterion.getItemSubtype(), -1);
+
+		DLFileEntryType dlFileEntryType = null;
+
+		if (fileEntryTypeId >= 0) {
+			dlFileEntryType = _dlFileEntryTypeLocalService.fetchFileEntryType(
+				fileEntryTypeId);
+		}
+
+		if (dlFileEntryType == null) {
+			fileEntryTypeId = _getFileEntryTypeId(
+				infoItemItemSelectorCriterion.getItemSubtype());
+		}
+
+		return fileEntryTypeId;
+	}
+
+	private long _getFileEntryTypeId(String itemSubtype) {
+		DLFileEntryType dlFileEntryType =
+			_dlFileEntryTypeLocalService.fetchFileEntryType(
+				_themeDisplay.getScopeGroupId(), itemSubtype);
+
+		if (dlFileEntryType == null) {
+			dlFileEntryType = _dlFileEntryTypeLocalService.fetchFileEntryType(
+				_themeDisplay.getCompanyGroupId(), itemSubtype);
+		}
+
+		if (dlFileEntryType != null) {
+			return dlFileEntryType.getFileEntryTypeId();
+		}
+
+		return 0;
```

Comment:
We can return here the constant `DLFileEntryTypeConstants.FILE_ENTRY_TYPE_ID_BASIC_DOCUMENT`.

Reply:
Done